src/main/kotlin/com/spectralogic/bp/bench/cli/AzSeekableByteChannel.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * ****************************************************************************
+ *   Copyright 2014-2019 Spectra Logic Corporation. All Rights Reserved.
+ * ***************************************************************************
+ */
+
+package com.spectralogic.bp.bench.cli
+
+import java.nio.ByteBuffer
+import java.nio.channels.SeekableByteChannel
+
+class AzSeekableByteChannel : SeekableByteChannel {
+
+    private var size = 0L
+    private var position = 0L
+    private var open = true
+    private var char = 'a'
+
+    override fun read(dst: ByteBuffer): Int {
+        val buffer = dst.asCharBuffer()
+        val amountToRead = Math.min(buffer.remaining(), buffer.limit())
+        val generator = generateSequence {
+            val charToReturn = char
+
+            char = char.inc()
+            if (char == '{') {
+                char = 'a'
+            }
+            charToReturn
+        }
+        generator.take(amountToRead).forEach {
+            size += 2
+            position += 2
+            buffer.put(it)
+        }
+
+        return amountToRead * 2
```

Comment:
Why is the amount read *2?

Reply:
The API says it should return the bytes read, but each character is 2 bytes long.

Reply:
I don't think that's strictly true.